std-bits/table/src/main/java/org/enso/table/data/column/storage/Storage.java

```diff
@@ -29,6 +29,13 @@
   /** @return the type tag of this column's storage. */
   public abstract StorageType getType();
 
+  /**
+   * @return the type of the values in this column's storage. Resolves type for Mixed if possible.
+   */
+  public StorageType getValuesType() {
```

Comment:
Personally I'd call this something like `inferPreciseType()` or something along these lines.